execution-engine/engine-core/src/execution/executor.rs

```diff
@@ -372,21 +374,19 @@ impl Executor<Module> for WasmiExecutor {
             Err(error) => error,
             Ok(_) => {
                 // This duplicates the behavior of sub_call, but is admittedly rather questionable.
-                let ret = bytesrepr::deserialize(runtime.result())?;
-                return Ok(ret);
+                return Ok(runtime.result().as_ref().cloned().unwrap_or(Value::Unit));
             }
         };
 
-        let return_value_bytes: &[u8] = match return_error
+        let return_value: Value = match return_error
             .as_host_error()
             .and_then(|host_error| host_error.downcast_ref::<Error>())
         {
-            Some(Error::Ret(_)) => runtime.result(),
+            Some(Error::Ret(_)) => runtime.result().as_ref().cloned().unwrap_or(Value::Unit),
```

Comment:
```suggestion
            Some(Error::Ret(_)) => runtime.result().to_owned().unwrap_or(Value::Unit),
```

(and other places)

but also, see other comment about `runtime.result`

Reply:
Fixed